fedimint-core/src/config.rs

```diff
@@ -198,16 +198,30 @@ impl ClientConfig {
     }
 
     /// Get the value of a given meta field
-    pub fn meta<V: serde::de::DeserializeOwned>(
+    pub fn meta<V: serde::de::DeserializeOwned + 'static>(
         &self,
         key: &str,
     ) -> Result<Option<V>, anyhow::Error> {
         let Some(str_value) = self.global.meta.get(key) else {
             return Ok(None);
         };
-        serde_json::from_str(str_value)
+        let res = serde_json::from_str(str_value)
             .map(Some)
-            .context(format!("Decoding meta field '{key}' failed"))
+            .context(format!("Decoding meta field '{key}' failed"));
+
+        // In the past we encoded some string fields as "just a string" without quotes,
+        // this code ensures that old meta values still parse since config is hard to
+        // change
+        if res.is_err() && std::any::TypeId::of::<V>() == std::any::TypeId::of::<String>() {
+            let string_ret = Box::new(str_value.clone());
+            let ret: Box<V> = unsafe {
+                // We can transmute a String to V because we know that V==String
+                std::mem::transmute(string_ret)
+            };
+            Ok(Some(*ret))
```

Comment:
I wasn't able to do this with a plain `V`, the compiler complained about unknown size, this seemed like the easiest fix.

Reply:
(@dpc used a similar unsafe trick elsewhere iirc, it's not new)

Reply:
> (@dpc used a similar unsafe trick elsewhere iirc, it's not new)

Yep, there's some insightful discussion related to this pattern in the PR that introduced it https://github.com/fedimint/fedimint/pull/4145

This looks good to me, but I'll defer approval to folks with more context.